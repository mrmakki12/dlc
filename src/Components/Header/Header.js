import React from "react";

export const Header = () => {

    return (
        <div className="container-fluid" data-testid="header">
            <div className="d-flex justify-content-between">
                <div>
                    <p>GameStation Store</p>
                </div>
                <div>
                    <i className="fa-solid fa-cart-shopping"></i>
                </div>
            </div>
            <div className=" text-white bg-dark row py-lg-4">
                {/* <img src="background.jpg" /> */}
                <div className="">
                    <h1 className="">Hiking Simulator</h1>
                    <h2 className="c">Add-Ons</h2>
                </div>
            </div>
        </div>
    )
}