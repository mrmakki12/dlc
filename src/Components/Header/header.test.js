import React from 'react';
import ReactDOM from 'react-dom';
import { Header } from './Header.js';
import { cleanup, render, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';

// cleanup
afterEach(cleanup);

// render Header 
test("renders Header without crashing", () => {
    const div  = document.createElement("div");
    ReactDOM.render(<Router><Header /></Router>, div)
});

// render Header
test("renders Header the right way", () => {
    const { getByTestId } = render(<Router><Header /></Router>);
    expect(getByTestId('header'));
});

// navigate to cart
test("navigate to Cart through user interaction", ()  => {
    // render header
    render(<Router><Header /></Router>);

    // click cart
    fireEvent.click(document.getElementsByTagName('i').item(0));

    // expect path to be cart
    expect(document.location.pathname).toBe('/cart');

});

// snapshot
test("matches snapshot", () => {
    const tree = renderer.create(<Router><Header /></Router>).toJSON();
    expect(tree).toMatchSnapshot();
});
