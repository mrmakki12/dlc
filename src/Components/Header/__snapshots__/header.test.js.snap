// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="container-fluid"
  data-testid="header"
>
  <div
    className="d-flex justify-content-between"
  >
    <div>
      <p>
        GameStation Store
      </p>
    </div>
    <a
      href="/cart"
      onClick={[Function]}
    >
      <div
        title="cart"
      >
        <i
          className="fa-solid fa-cart-shopping"
        />
      </div>
    </a>
  </div>
  <div
    className=" text-white bg-dark row py-lg-4"
  >
    <div
      className=""
    >
      <h1
        className=""
      >
        Hiking Simulator
      </h1>
      <h2
        className="c"
      >
        Add-Ons
      </h2>
    </div>
  </div>
</div>
`;
